kernel/src/mm/phys/mod.rs

```diff
@@ -64,6 +65,37 @@ static mut FRAME_ALLOCATOR_STORAGE: [u8; config::kernel::MEMORY_SIZE
 // Standalone Functions
 //==================================================================================================
 
+fn parse_physical_memory_regions(
+    physical_memory_regions: LinkedList<TruncatedMemoryRegion<PhysicalAddress>>,
+) -> Result<
+    (
+        LinkedList<TruncatedMemoryRegion<PhysicalAddress>>,
+        LinkedList<TruncatedMemoryRegion<PhysicalAddress>>
+    ),
+    Error,
+> {
+    let mut usable_physical_memory_regions: LinkedList<TruncatedMemoryRegion<PhysicalAddress>> =
+        LinkedList::new();
+    let mut reserved_physical_memory_regions: LinkedList<TruncatedMemoryRegion<PhysicalAddress>> =
+        LinkedList::new();
+
+    for region in physical_memory_regions {
+        match region.typ() {
+            MemoryRegionType::Usable => {
+                usable_physical_memory_regions.push_back(region);
+            },
+            MemoryRegionType::Reserved => {
+                reserved_physical_memory_regions.push_back(region);
+            }
+            _ => {
+                return Err(Error::new(ErrorCode::InvalidArgument, "Invalid memory type"));
```

Comment:
Could you please exapnd this to follow a pattern where we log the error message? Something like:

```rust
let reason: &str = "invalid memory type"
error!("parse_physical_memory_regions(): {:?}", reason)
return Err(Error::new(ErrorCode::InvalidArgument, &reason));
```

Reply:
Done.